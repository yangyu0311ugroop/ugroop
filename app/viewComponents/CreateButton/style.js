const styles = {
  root: {
    marginBottom: -10,
  },
};

export default styles;
