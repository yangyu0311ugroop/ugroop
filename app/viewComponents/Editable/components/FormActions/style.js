const style = {
  item: {
    right: 0,
    bottom: -42,
    zIndex: 99,
    display: 'flex',
    position: 'absolute',
    left: 0,
  },
  inline: {
    bottom: 'unset',
    top: -32,
  },
  submit: {
    marginRight: 4,
  },
  secondaryRoot: {
    display: 'flex',
    alignItems: 'center',
    '&::after': {
      content: "' '",
      width: 1,
      height: 30,
      margin: '0 4px',
      border: '1px dashed #858585',
    },
  },
};

export default style;
