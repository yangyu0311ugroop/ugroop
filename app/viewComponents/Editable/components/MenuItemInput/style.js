const style = {
  root: {
    paddingBottom: 8,
  },
};

export default style;
