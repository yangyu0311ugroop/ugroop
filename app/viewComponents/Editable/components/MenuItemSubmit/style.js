const style = {
  root: {
    backgroundColor: '#f7f9ff',
  },
  disabled: {
    backgroundColor: '#fff7fb',
  },
};

export default style;
