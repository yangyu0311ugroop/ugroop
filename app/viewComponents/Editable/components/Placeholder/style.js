const style = {
  p: {
    color: '#8c8c8c',
    fontStyle: 'italic',
  },
};

export default style;
