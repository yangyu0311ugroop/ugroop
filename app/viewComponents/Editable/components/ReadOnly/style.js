import { scalingFontSizes } from 'theme/commonStyles/scalingFontSizes';

const style = theme => ({
  ...scalingFontSizes(theme, false),

  bold: {
    fontWeight: 600,
  },

  root: {
    display: 'block',
    position: 'relative',
  },
});

export default style;
