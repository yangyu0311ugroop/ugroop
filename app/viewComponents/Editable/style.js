/**
 * Created by stephenkarpinskyj on 20/7/18.
 */

const style = {
  root: {
    margin: '-2px -2px',
    borderRadius: 4,
  },

  fullWidth: {
    flex: 1,
  },

  content: {
    padding: '0 2px',
    borderRadius: 4,
    display: 'flex',
    color: 'unset',
    fontSize: 'unset',
    position: 'relative',
    alignItems: 'center',
    userSelect: 'text',
    '&:hover': {
      backgroundColor: '#E0E8ED',
    },
  },
  label: {
    textAlign: 'left',
  },
  hover: {
    paddingTop: 4,
    transition: '.2s ease-in-out',
  },

  contentHover: {
    backgroundColor: '#ebecf0',
    '&:hover': {
      backgroundColor: '#ebecf0',
    },
  },

  icon: {
    right: -32,
    zIndex: 1,
    padding: 2,
    marginLeft: 8,
    borderRadius: 4,
    height: '100%',
    minWidth: 24,
    display: 'none',
    position: 'absolute',
  },
  iconHover: {
    display: 'initial',
  },
  displayFlex: {
    display: 'flex',
  },
};

export default style;
