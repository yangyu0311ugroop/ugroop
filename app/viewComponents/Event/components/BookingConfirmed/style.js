const style = {
  iconRoot: {
    position: 'absolute',
    zIndex: 2,
    top: 4,
    right: 5,
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    height: 10,
    width: 10,
  },

  cardRoot: {
    fontWeight: '600',
    color: '#7aa1ea',
    fontSize: '12px',
  },
  checkmark: {
    fontWeight: '400',
    fontSize: '20px',
  },
};

export default style;
