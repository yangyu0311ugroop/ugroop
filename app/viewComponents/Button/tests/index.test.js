import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import coolTheme from 'theme/coolTheme';
import mockStylesheet from 'utils/mockStylesheet';
import style from '../styles';
import { Button } from '../index';

describe('Button/tests/index.test.js', () => {
  const children = <span>Title</span>;
  let rendered;

  beforeEach(() => {
    rendered = shallow(<Button classes={{}}>{children}</Button>);
  });

  describe('Smoke tests', () => {
    it('should exist', () => {
      expect(Button).toBeDefined();
    });
    it('should render without exploding', () => {
      expect(rendered.length).toBe(1);
    });
    it('should render properly', () => {
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('style snapshot', () => {
    it('should match snapshot', () => {
      const classes = mockStylesheet('Button', style, coolTheme);
      expect(classes).toMatchSnapshot();
    });
  });

  describe('icon button', () => {
    it('should render icon button', () => {
      rendered.setProps({ iconButton: true, icon: 'folder' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render small icon', () => {
      rendered.setProps({ iconButton: true, icon: 'folder', small: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render large icon', () => {
      rendered.setProps({ iconButton: true, icon: 'folder', large: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('button with tooltip', () => {
    it('should render a tooltip', () => {
      const tooltipProps = {
        title: 'Title',
        placement: 'top',
      };
      rendered.setProps({ tooltipProps });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should not render a tooltip', () => {
      const tooltipProps = {};
      rendered.setProps({ tooltipProps });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('buttonStyle', () => {
    it('should render inline', () => {
      rendered.setProps({ variant: 'inline' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render transparent outline', () => {
      rendered.setProps({ variant: 'outline', transparent: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render borderless', () => {
      rendered.setProps({ variant: 'borderless' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('button color', () => {
    it('should render primary color', () => {
      rendered.setProps({ color: 'primary' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render alert color', () => {
      rendered.setProps({ color: 'alert' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render gray color', () => {
      rendered.setProps({ color: 'gray' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render gray color', () => {
      rendered.setProps({ color: 'black' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('button size', () => {
    it('should render small size', () => {
      rendered.setProps({ size: 'small' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render large size', () => {
      rendered.setProps({ size: 'large' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('disabled button', () => {
    it('should be disabled', () => {
      rendered.setProps({ disabled: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('square button', () => {
    it('should be square', () => {
      rendered.setProps({ square: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('dense button', () => {
    it('should be densed', () => {
      rendered.setProps({ dense: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('block button', () => {
    it('should be block', () => {
      rendered.setProps({ block: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });

  describe('loading button', () => {
    it('should render loading button', () => {
      rendered.setProps({ loading: true });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render outlineBase or borderlessBase loading button', () => {
      rendered.setProps({ loading: true, variant: 'outline' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render outlinePrimary or borderlessPrimary loading button', () => {
      rendered.setProps({
        loading: true,
        variant: 'outline',
        color: 'primary',
      });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render outlineAlert or borderlessAlert loading button', () => {
      rendered.setProps({ loading: true, variant: 'outline', color: 'alert' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
    it('should render outlineGray or borderlessGray loading button', () => {
      rendered.setProps({ loading: true, variant: 'outline', color: 'gray' });
      expect(toJSON(rendered)).toMatchSnapshot();
    });
  });
});
