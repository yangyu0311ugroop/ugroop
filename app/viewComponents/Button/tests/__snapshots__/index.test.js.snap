// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button/tests/index.test.js Smoke tests should render properly 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js block button should be block 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button color should render alert color 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button color should render gray color 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button color should render gray color 2`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button color should render primary color 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button size should render large size 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button size should render small size 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button with tooltip should not render a tooltip 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js button with tooltip should render a tooltip 1`] = `
<WithStyles(Tooltip)
  enterDelay={700}
  isLight={true}
  placement="top"
  title="Title"
  tooltipProps={
    Object {
      "rest": Object {
        "placement": "top",
      },
    }
  }
>
  <WithStyles(ForwardRef(Button))
    className=""
    disableRipple={false}
    disabled={false}
  >
    <span>
      Title
    </span>
  </WithStyles(ForwardRef(Button))>
</WithStyles(Tooltip)>
`;

exports[`Button/tests/index.test.js buttonStyle should render borderless 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js buttonStyle should render inline 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js buttonStyle should render transparent outline 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js dense button should be densed 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js disabled button should be disabled 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={true}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js icon button should render icon button 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
  icon="folder"
>
  <WithStyles(Icon)
    className=""
    icon="folder"
    size="base"
  />
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js icon button should render large icon 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
  icon="folder"
  large={true}
>
  <WithStyles(Icon)
    className=""
    icon="folder"
    size="base"
  />
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js icon button should render small icon 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
  icon="folder"
  small={true}
>
  <WithStyles(Icon)
    className=""
    icon="folder"
    size="base"
  />
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js loading button should render loading button 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={true}
  disabled={true}
>
  <WithStyles(Icon)
    className=""
    icon="loading"
    size="base"
  />
  <div
    className="undefined"
  >
    <span>
      Title
    </span>
  </div>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js loading button should render outlineAlert or borderlessAlert loading button 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={true}
  disabled={true}
>
  <WithStyles(Icon)
    className=""
    icon="loading"
    size="base"
  />
  <div
    className="undefined"
  >
    <span>
      Title
    </span>
  </div>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js loading button should render outlineBase or borderlessBase loading button 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={true}
  disabled={true}
>
  <WithStyles(Icon)
    className=""
    icon="loading"
    size="base"
  />
  <div
    className="undefined"
  >
    <span>
      Title
    </span>
  </div>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js loading button should render outlineGray or borderlessGray loading button 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={true}
  disabled={true}
>
  <WithStyles(Icon)
    className=""
    icon="loading"
    size="base"
  />
  <div
    className="undefined"
  >
    <span>
      Title
    </span>
  </div>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js loading button should render outlinePrimary or borderlessPrimary loading button 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={true}
  disabled={true}
>
  <WithStyles(Icon)
    className=""
    icon="loading"
    size="base"
  />
  <div
    className="undefined"
  >
    <span>
      Title
    </span>
  </div>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js square button should be square 1`] = `
<WithStyles(ForwardRef(Button))
  className=""
  disableRipple={false}
  disabled={false}
>
  <span>
    Title
  </span>
</WithStyles(ForwardRef(Button))>
`;

exports[`Button/tests/index.test.js style snapshot should match snapshot 1`] = `
Object {
  "@keyframes spin": "Button-@keyframes spin",
  "base": "Button-base",
  "baseIcon": "Button-baseIcon",
  "block": "Button-block",
  "bold": "Button-bold",
  "borderless": "Button-borderless",
  "borderlessAlert": "Button-borderlessAlert",
  "borderlessBase": "Button-borderlessBase",
  "borderlessGray": "Button-borderlessGray",
  "borderlessPending": "Button-borderlessPending",
  "borderlessPrimary": "Button-borderlessPrimary",
  "borderlessYellow": "Button-borderlessYellow",
  "buttonText": "Button-buttonText",
  "dense": "Button-dense",
  "dialog": "Button-dialog",
  "extraSmall": "Button-extraSmall",
  "extraSmallIcon": "Button-extraSmallIcon",
  "iconRoot": "Button-iconRoot",
  "inline": "Button-inline",
  "large": "Button-large",
  "largeIcon": "Button-largeIcon",
  "light": "Button-light",
  "loading": "Button-loading",
  "loadingChild": "Button-loadingChild",
  "loadingSpan": "Button-loadingSpan",
  "noBorderRadius": "Button-noBorderRadius",
  "noBoxShadow": "Button-noBoxShadow",
  "noMargin": "Button-noMargin",
  "noPadding": "Button-noPadding",
  "normal": "Button-normal",
  "outline": "Button-outline",
  "outlineAlert": "Button-outlineAlert",
  "outlineBase": "Button-outlineBase",
  "outlineBlack": "Button-outlineBlack",
  "outlineDarkgray": "Button-outlineDarkgray",
  "outlineGray": "Button-outlineGray",
  "outlinePending": "Button-outlinePending",
  "outlinePrimary": "Button-outlinePrimary",
  "outlineYellow": "Button-outlineYellow",
  "root": "Button-root",
  "small": "Button-small",
  "smallIcon": "Button-smallIcon",
  "square": "Button-square",
  "standard": "Button-standard",
  "standardAlert": "Button-standardAlert",
  "standardBase": "Button-standardBase",
  "standardBlack": "Button-standardBlack",
  "standardGray": "Button-standardGray",
  "standardInline": "Button-standardInline",
  "standardNormal": "Button-standardNormal",
  "standardPending": "Button-standardPending",
  "standardPrimary": "Button-standardPrimary",
  "standardWhite": "Button-standardWhite",
  "standardXs": "Button-standardXs",
  "strong": "Button-strong",
  "textAlignCenter": "Button-textAlignCenter",
  "textAlignLeft": "Button-textAlignLeft",
  "textAlignRight": "Button-textAlignRight",
  "transparent": "Button-transparent",
  "verySquare": "Button-verySquare",
  "xs": "Button-xs",
  "xxs": "Button-xxs",
}
`;
